### PR TITLE
Migrated colors that used in theme_properties to color provider pipeline

### PR DESCRIPTION
**Pull request by simonhong**
https://github.com/brave/brave-browser/issues/23423

Introduced for color mixers for light/dark/private/tor window for using
our theme colors from color provider.

With this PR, brave's background color of new tab button/tab search button
is used.

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

`npm run test brave_browser_tests --filter=BraveThemeServiceTest.ColorProviderTest`

1. Launch browser and check the background of new tab button in all windows(light, dark, private, private tor)
2. Install any external theme extension from web store and check proper new tab button color is applied

### CODE REVIEW

**Review by emerick — APPROVED** [review]
LGTM

**Review by goodov — APPROVED** [review]

File: browser/ui/color/sources.gni [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2022 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+brave_browser_ui_colors_sources = [
```
Comment (goodov) [inline browser/ui/color/sources.gni:6]: nit: I think it should be `brave_browser_ui_color_mixers_sources` (and var below fixed accordingly).

File: browser/ui/color/color_palette.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_COLOR_COLOR_PALETTE_H_
+#define BRAVE_BROWSER_UI_COLOR_COLOR_PALETTE_H_
+
+#include "third_party/skia/include/core/SkColor.h"
+
+constexpr SkColor kLightToolbar = SkColorSetRGB(0xf3, 0xf3, 0xf3);
```
Comment (petemill) [inline browser/ui/color/color_palette.h:11]: Why put these in a separate file and not in an anonymous namespace in brave_color_mixer directly? What is the rule of which colors belong here and which are specified directly in brave_color_mixer? E.g. `kColorBookmarkBarForeground` and `kColorTabForegroundActiveFrameActive` are specified directly.

I can imagine making a `color_palette.h` in `ui/` that mirrors the new Brave design system (and we will generate it in https://github.com/brave-experiments/leo) but I'm not sure if having a separation of these browser-specific colors is confusing when they are only referenced in 1 class?

**Review by petemill — APPROVED** [review]
This is great and straightforward to read - thanks for putting the work in. For the values, it seems like a port from existing values so we should be good. I just left a note about structure but it's just a nit or a question, and up to you!

File: browser/ui/color/sources.gni (hunk 1 above)
Comment (simonhong) [reply browser/ui/color/sources.gni]: Renamed.

File: browser/ui/color/color_palette.h (hunk 2 above)
Comment (simonhong) [reply browser/ui/color/color_palette.h:15]: You're right. I left comment here like below.
```
// TODO(simonhong): Move all below colors to brave_color_mixer.cc.
// Originally this colors were defined in BraveThemeProperties but
// they are used from mixers also. When color fetching from theme
// provider is deprecated, only mixer will use these colors.
```

**Review by simonhong — COMMENTED** [review]
I realized that this PR works nicely on Windows/Mac but not on Linux.
My test case detected :). Looking on linux.

**Review by sangwoo108 — APPROVED** [review]
LGTM

File: browser/ui/views/frame/brave_browser_frame.cc [hunk 3]
```diff
@@ -28,3 +37,12 @@ const ui::NativeTheme* BraveBrowserFrame::GetNativeTheme() const {
   }
   return views::Widget::GetNativeTheme();
 }
+
+ui::ColorProviderManager::ThemeInitializerSupplier*
+BraveBrowserFrame::GetCustomTheme() const {
+  if (theme_supplier_) {
+    return theme_supplier_.get();
+  }
```
[the remark this answers is not in the document]
Comment (simonhong) [reply browser/ui/views/frame/brave_browser_frame.cc:50]: Added.

**Comment by simonhong** [comment]
Pushed another commit for using BravePrivateWindowThemeSupplier on linux.

**Review by sangwoo108 — APPROVED** [review]
Still LGTM

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.42.20 Chromium: 103.0.5060.53 (Official Build) nightly (64-bit)
-- | --
Revision | a1711811edd74ff1cf2150f36ffa3b0dae40b17f-refs/branch-heads/5060@{#853}
OS | Windows 11 Version 21H2 (Build 22000.739)
```

Went through the STR/Cases outlined via https://github.com/brave/brave-core/pull/13841#issue-1275035034

### `Light Theme`

**`Before Fix`** | **`After Fix`**
-----------------|-----------------
beforeMainLight | afterMainLight
beforePBLight | afterPBLight
beforeTorLight | afterTorLight

### `Dark Theme`

**`Before Fix`** | **`After Fix`**
-----------------|-----------------
beforeMainDark | afterMainDark
beforePBDark | afterPBDark
beforeTorDark | afterTorDark

### `Custom Themes`

With the following checks, basically ensured that the `+` looked as expected with the custom theme.

**`Before Fix`** | **`After Fix`**
-----------------|-----------------
beforeCustom1 | afterCustom1
beforeCustom2 | afterCustom2

**Pull request merged**